### PR TITLE
lazy-load

### PR DESCRIPTION
**Pull request by vadimdom**
test

### CODE REVIEW

File: server/Domain/Services/Impl/PostService.ts [hunk 1]
```diff
@@ -10,8 +10,14 @@ export class PostService implements IPostService {
...
-    return this._postRepository.findAll({});
+  public async getPosts(count: number, offset: number): Promise<PostEntity[]> {
+    if (!count && !offset) {
+      return this._postRepository.findAll({});
+    } else if (!offset) {
+      return this._postRepository.findAll({ limit: +count });
+    } else {
+      return this._postRepository.findAll({ limit: +count, offset: +offset });
+    }
   }
```
Comment (vadimdom) [inline server/Domain/Services/Impl/PostService.ts]: if else need?

**Comment by Rezdoom** [comment]
remove `nextFn` from imports if it is not used

File: server/API/controllers/PostController.ts [hunk 2]
```diff
@@ -62,33 +48,16 @@ export class PostController implements interfaces.Controller {
...
-    @requestParam() data: any,
-    @response() res: express.Response,
-    @nextFn() next: express.NextFunction
-  ): Promise<void> {
-    try {
-      res.json(await this._postService.deletePost(data.id));
-    } catch (error) {
-      next(error);
-    }
+  private async delete(@requestParam('idPost') idPost: number, @response() res: express.Response): Promise<void> {
```
Comment (Rezdoom) [inline server/API/controllers/PostController.ts]: request param has only `id` and not `idPost`, rename

File: server/API/controllers/PostController.ts [hunk 3]
```diff
@@ -62,33 +48,16 @@ export class PostController implements interfaces.Controller {
...
+    res.json(await this._postService.deletePost(res.req.params.id));
```
Comment (Rezdoom) [inline server/API/controllers/PostController.ts]: WTF? res.req? change to `id` from `requstParam`

Comment (Rezdoom) [reply server/API/controllers/PostController.ts]: and what motivated you to change it?

File: server/API/controllers/PostController.ts [hunk 4]
```diff
@@ -30,30 +29,17 @@ export class PostController implements interfaces.Controller {
...
   @httpGet('/')
-  // @ts-ignore
-  private async get(@response() res: express.Response, @nextFn() next: express.NextFunction): Promise<void> {
-    try {
-      res.json(await this._postService.getPosts());
-    } catch (error) {
-      next(error);
-    }
+  private async get(@response() res: express.Response): Promise<void> {
+    res.json(await this._postService.getPosts(res.req.query.count, res.req.query.offset));
```
Comment (Rezdoom) [inline server/API/controllers/PostController.ts]: res.req????
For all params of expressjs middleware  in `inversify-express-utils` there are decorators. `req` is `@request`
But in `inversify-express-utils`  you can get value from query string via `@queryParam('name')`

File: server/API/controllers/PostController.ts [hunk 5]
```diff
@@ -62,33 +48,16 @@ export class PostController implements interfaces.Controller {
...
-    }
+  private async updatePost(@requestBody() data: any, @response() res: express.Response): Promise<void> {
+    res.json(await this._postService.updatePost(data));
   }
 
   /**
    * Delete post
    * id: post's id
    */
   @httpDelete('/:id')
```
Comment (Rezdoom) [inline server/API/controllers/PostController.ts]: id here, not idPost

**Pull request merged**